### PR TITLE
Charl

### PR DESCRIPTION
**Pull request by charl26**

### CODE REVIEW

File: src/app/_components/login/login.component.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  text-align: center;
+  max-width: 920px;
+}
+
+form {
+
+  margin: 10%;
+  border-radius: 5px;
+  padding: 16px;
+  background-color: #e1e1e1;
```
Comment (DJRave) [inline src/app/_components/login/login.component.scss]: We should use a variable file to hold all the colors

File: src/app/_components/login/login.component.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+mat-form-field {
+  margin: 8px;
```
Comment (DJRave) [inline src/app/_components/login/login.component.scss]: should not use px as this becomes very hard to work with better to use rem/em

File: src/app/_components/login/login.component.scss (hunk 1 above)
Comment (charl26) [reply src/app/_components/login/login.component.scss]: Okay I'm not sure how to do that could you show me

File: src/app/_components/login/login.component.scss (hunk 2 above)
Comment (charl26) [reply src/app/_components/login/login.component.scss]: okay will change

File: src/app/_services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	constructor(private router: Router) {
+	}
+
+	get isLoggedIn() {
+		return this.loggedIn.asObservable();
+	}
+
+	login(user: User) {
+		if (user.userName !== '' && user.password !== '') {
```
Comment (DJRave) [inline src/app/_services/auth.service.ts]: better to use the double bang here to make sure that user.username is not null or undefined

File: src/app/_components/login/login.component.scss (hunk 2 above)
Comment (charl26) [reply src/app/_components/login/login.component.scss]: should i changed it on everywhere i've add px

File: src/app/app-routing.module.ts [hunk 4]
```diff
@@ -2,16 +2,28 @@ import { NgModule } from '@angular/core';
...
+import { AuthGuard } from './_guards/auth.guard';
 
 const routes: Routes = [
-    {path: '', redirectTo: '/login', pathMatch: 'full'},
+	{path: '', redirectTo: '/login', pathMatch: 'full'},
 
-    {path: 'login', component: LoginComponent},
+	{path: 'login', component: LoginComponent},
+
+	{path: 'schedules', component: SchedulesComponent, canActivate: [AuthGuard]},
```
Comment (DJRave) [inline src/app/app-routing.module.ts]: Question are we going to have lazy loading?

Comment (charl26) [reply src/app/app-routing.module.ts]: which would be better

File: src/styles.scss [hunk 5]
```diff
@@ -3,12 +3,13 @@
...
-
 body {
   font-family: Roboto, Arial, sans-serif;
+  text-align: -webkit-center;
   margin: 0;
 }
+
+.mat-toolbar {
+  background: #191919;
+  border-radius: 0 !important;
```
Comment (DJRave) [inline src/styles.scss]: Please do not use important when you don't have to if there is no way around it leave a commit why you need to use important

Comment (charl26) [reply src/styles.scss]: okay will do

File: src/app/_components/login/login.component.scss (hunk 2 above)
Comment (DJRave) [reply src/app/_components/login/login.component.scss]: Yes please so the math is px/12* 1 rem or em where rem is the parent container and em is the current container

File: src/app/_services/auth.service.ts (hunk 3 above)
Comment (charl26) [reply src/app/_services/auth.service.ts]: I've never used a double bang will have a look at it

File: src/app/_components/login/login.component.scss (hunk 1 above)
Comment (DJRave) [reply src/app/_components/login/login.component.scss]: will do

File: src/app/app-routing.module.ts (hunk 4 above)
Comment (DJRave) [reply src/app/app-routing.module.ts]: depends on the size of the app if you have a med-large app you will use lazy loading to reduce to size on load and then the client will only load what they have to

**Pull request merged**